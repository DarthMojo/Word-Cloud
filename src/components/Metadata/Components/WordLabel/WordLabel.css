.WordLabel {
    font-size: 4em;
    cursor: pointer;
    display: flex;
    justify-content: center;
}