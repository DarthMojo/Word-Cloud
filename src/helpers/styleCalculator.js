/**
 * 1. Each topic should have one of 6 different text sizes,
 * with the most popular topics largest, and least popular smallest
 *
 * 2.A topic with a sentiment score > 60 should be displayed in green
 *
 * 3.A topic with a sentiment score < 40 should be displayed in red
 *
 * 4.Other topics should be displayed in grey
 */
import type Dimension from "../config/types";
import { sentimentScoreMin,
    sentimentScoreMax,
    defaultColor,
    size } from "../config/static";


/**
 *
 * @type {{getTopicSize: (function(*, *)),
 * getTopicColor: (function(*)),
 * windowSize: (function(*=)),
 * rotation: (function(): number)}}
 */
const styleCalculator = {
    /**
     *
     * @param wordPopularity
     * @param popularityRange
     * @return {number}
     */
    getTopicSize: (wordPopularity, popularityRange): number => {
        const unitRange = ((popularityRange.max - popularityRange.min) / size.max) || 0;
        const sizeValue: number = Array.from({ length: size.max }, (v, k): number => k + 1)
            .find((sizeIndex: number): number =>
                wordPopularity <= popularityRange.min + (sizeIndex * unitRange));
        return Math.round((sizeValue / size.max) * 100);
    },
    /**
     *
     * @param sentimentScore
     * @return String
     */
    getTopicColor: (sentimentScore): string => {
        if (sentimentScore > sentimentScoreMax.score) {
            return sentimentScoreMax.color;
        } else if (sentimentScore < sentimentScoreMin.score) {
            return sentimentScoreMin.color;
        }
        return defaultColor;
    },
    /**
     *
     * @param factor
     * @return {{width: number, height: number}}
     */
    windowSize: (factor = 1): Dimension => {
        return {
            width: Math.round(window.innerWidth * factor),
            height: Math.round(window.innerHeight * factor)
        };
    },
    /**
     *@return {number}
     */
    rotation: (): number => (~~(Math.random() * size.max) - size.max * 0.5) * 30
};
export default styleCalculator;
