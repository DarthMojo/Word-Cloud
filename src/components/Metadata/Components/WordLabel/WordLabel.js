import React from "react";
import type { Element } from "react";

import "./WordLabel.css";

type WordLabelType = {
    wordColor: string;
    wordLabel: string;
    dimensions: Object;
};

/**
 *
 * @param wordColor
 * @param wordLabel
 * @param dimensions
 * @param onWordLabelClick
 * @return Element
 */
const WordLabel = ({ wordColor, wordLabel, dimensions }: WordLabelType): Element<any> => (
    <div className="WordLabel"
        style={{ height: dimensions.height * 0.5, width: dimensions.width }}>
        <div
            style={{ color: wordColor }}
        > {wordLabel}
        </div>
    </div>);

export default WordLabel;
