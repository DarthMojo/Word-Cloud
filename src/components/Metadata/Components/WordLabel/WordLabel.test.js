import React from "react";
import styleCalculator from "../../../../helpers/styleCalculator";
import WordLabel from "./WordLabel";
import { shallowToJson } from "enzyme-to-json";

test("render WordLabel()", () => {
    const dimensions = styleCalculator.windowSize();

    const wrapper = shallow(
        <WordLabel
            dimensions={dimensions}
            wordColor={"black"}
            wordLabel={"mockLabel"}
        />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});
