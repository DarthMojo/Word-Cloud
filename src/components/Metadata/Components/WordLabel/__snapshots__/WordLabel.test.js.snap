// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render WordLabel() 1`] = `
<div
  className="WordLabel"
  style={
    Object {
      "height": 384,
      "width": 1024,
    }
  }
>
  <div
    style={
      Object {
        "color": "black",
      }
    }
  >
     
    mockLabel
  </div>
</div>
`;
